session.go

```diff
@@ -1059,30 +1056,24 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
 
 // txnFuture is a promise, which promises to return a txn in future.
 type txnFuture struct {
-	sync.WaitGroup
-	txn kv.Transaction
-	err error
+	future oracle.Future
+	store  kv.Storage
 }
 
-func (s *session) getTxnFuture() *txnFuture {
-	if s.txnFutureCh == nil {
-		s.txnFutureCh = make(chan *txnFuture, 1)
-		go asyncGetTSWorker(s.store, s.txnFutureCh)
+func (tf *txnFuture) wait() (kv.Transaction, error) {
+	startTS, err := tf.future.Wait()
+	if err == nil {
```

Comment:
This error is always `Canceled`, caused by network timeout.
The real reason may be pd leader change or server down or network problem...

But it doesn't matter, if err not nil, we'll call `store.Begin()` in line 1070, 
which would retry get timestamp until timeout.